test/jdk/java/awt/im/memoryleak/InputContextMemoryLeakTest.java

```diff
@@ -61,10 +61,12 @@ public void run() {
                 button = new JButton("Test");
                 p1.add(button);
                 frame.add(p1);
-                text = new WeakReference<JTextField>(new JTextField("Text"));
-                p = new WeakReference<JPanel>(new JPanel(new FlowLayout()));
-                p.get().add(text.get());
-                frame.add(p.get());
+                JTextField tf = new JTextField("Text");
+                text = new WeakReference<JTextField>(tf);
+                JPanel jp = new JPanel(new FlowLayout());
+                p = new WeakReference<JPanel>(jp);
+                jp.add(tf);
+                frame.add(jp);
```

Comment:
```
    text = new WeakReference<JTextField>(new JTextField("Text"));
    p = new WeakReference<JPanel>(new JPanel(new FlowLayout()));
    // assume that p.get() and text.get() are always non-null. What coulld go wrong?
    p.get().add(text.get());
```

Well oops. So this is the test failure fix you mentioned offline. This, and the below change, looks good.